cucumber-jparest/src/main/java/uk/gov/homeoffice/digital/sas/cucumberjparest/PayloadManager.java

```diff
@@ -0,0 +1,83 @@
+package uk.gov.homeoffice.digital.sas.cucumberjparest;
+
+import java.util.HashMap;
+import java.util.Map;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Setter;
+import org.springframework.stereotype.Component;
+
+@Component
+public class PayloadManager {
+
+  // Holds that state for payloads
+  private final Map<PayloadKey, String> payloads = new HashMap<>();
+
+  /**
+   * Creates a new payload with the given name.
+   *
+   * @param key     The key to use for the payload consisting of resource type and name
+   * @param content The content of the payload
+   */
+  public void createPayload(PayloadKey key, String content) {
+    if (payloads.containsKey(key)) {
+      throw new IllegalArgumentException(
+          "A payload with the name \"" + key.name + "\" already exists");
+    }
+
+    payloads.put(key, content);
+  }
+
+  /**
+   * Retrieves a payload added by
+   *     the {@link PayloadManager#createPayload(PayloadKey, String)} method.
+   *
+   * @param key The name used to reference the payload
+   * @return Payload
+   */
+  public String getPayload(PayloadKey key) {
+
+    if (!payloads.containsKey(key)) {
+      throw new IllegalArgumentException(
+          "A payload with the name \"" + key.name + "\" does not exist");
+    }
+    return payloads.get(key);
+  }
+
+  /**
+   * Used to hold a key for a defined payload. These take the form of the resource type and a
+   * variable that can be used to reference the payload throughout a scenario
+   */
+  @AllArgsConstructor
+  public static class PayloadKey {
+
+    @NonNull
+    @Getter
+    @Setter
+    private String resourceType;
+
+    @NonNull
+    @Getter
+    @Setter
+    private String name;
+
+    @Override
+    public final int hashCode() {
```

Comment:
does this need to be final? I think that would prevent us being able to override the hashcode method in a (future) subclass with additional fields...
Same for equals method.

Reply:
@stephenkaye-ho can you comment on this please?

Reply:
I've made made PayloadKey class final